### PR TITLE
[chore][processor/memorylimiter] Cleanup README

### PR DESCRIPTION
**Pull request by crobert-1**
**Documentation:** <Describe the documentation added.>
Some of the wording was hard to follow in the README so I attempted to make it more clear. There shouldn't be any real content changes.

### CODE REVIEW

File: processor/memorylimiterprocessor/README.md [hunk 1]
```diff
@@ -8,58 +8,60 @@
...
+The processor uses soft and hard memory limits. The hard limit is always above or equal
+to the soft limit.
 
-When the memory usage exceeds the soft limit the processor will enter the memory limited
-mode and will start refusing the data by returning errors to the preceding component
+The processor will enter memory limited mode and will start refusing the data when
+memory usage exceeds the soft limit. This is done by returning errors to the preceding component
 in the pipeline that made the ConsumeLogs/Trace/Metrics function call.
-The preceding component should be normally a receiver.
+The preceding component should normally be a receiver.
```
Comment (TylerHelmuth) [inline processor/memorylimiterprocessor/README.md:25]: It might be easiest to say "The best practice is to add the memory limiter processor as the first processor in a pipeline."

File: processor/memorylimiterprocessor/README.md [hunk 2]
```diff
@@ -8,58 +8,60 @@
...
-eventually result in dropped data since the receivers may not be able to hold back
-and retry the data indefinitely.
+eventually result in dropped data since the receivers may not be able to
+retry the data indefinitely.
 
-It is highly recommended to configure `ballastextension` as well as the
-`memory_limiter` processor on every collector. The ballast should be configured to
-be 1/3 to 1/2 of the memory allocated to the collector. The memory_limiter
+It is highly recommended to configure the `ballastextension` as well as the
+`memory_limiter` processor on every collector. The `ballastextension` should be configured to
```
Comment (TylerHelmuth) [inline processor/memorylimiterprocessor/README.md:60]: @mx-psi @dmitryax should we be recommending `GOMEMLIMIT` instead?

Comment (dmitryax) [reply processor/memorylimiterprocessor/README.md:60]: Right. We should

Comment (mx-psi) [reply processor/memorylimiterprocessor/README.md:60]: Indeed

File: processor/memorylimiterprocessor/README.md (hunk 1 above)
Comment (crobert-1) [reply processor/memorylimiterprocessor/README.md]: I found that this is stated twice in the README, so I've moved what you've suggested here to be where the second mention of this best practice was.

File: processor/memorylimiterprocessor/README.md (hunk 2 above)
Comment (crobert-1) [reply processor/memorylimiterprocessor/README.md]: I've updated based on the recommendation in the deprecation notice of the ballast extension. Feedback is appreciated to ensure it's correct.

**Comment by crobert-1** [comment]
I added some headers in an attempt to organize the README a bit. If this is distracting or too far from the standard I'm happy to revert.

**Review by mx-psi — APPROVED** [review]

File: processor/memorylimiterprocessor/README.md [hunk 3]
```diff
@@ -12,61 +12,72 @@
...
+the same data. The receivers may also apply backpressure to their own data sources
+in order to slow the inflow of data into the Collector, and to allow memory usage
+to go below the set limits.
 
->Warning: if the component preceding the memory limiter in the pipeline does not correctly
-retry and send the data again after ConsumeLogs/Trace/Metrics functions return then that
-data will be permanently lost. We consider such components incorrectly implemented.
+>Warning: Data will be permanently lost if the component preceding the memory limiter
+> in the telemetry pipeline does not correctly retry sending data after the
+> ConsumeLogs/Trace/Metrics function returns.
```
Comment (TylerHelmuth) [inline processor/memorylimiterprocessor/README.md:46]: I dont expect users of the memory limiter to understand what these functions mean.  Can we reword this to something like "does not correctly retry sending data after the memory limiter processor refuses data."

Comment (crobert-1) [reply processor/memorylimiterprocessor/README.md]: Good call, describing functionality is definitely easier to understand than referencing function names. 👍

**Review by TylerHelmuth — APPROVED** [review]
